text/0000-otlp-over-messaging-systems.md

```diff
@@ -0,0 +1,110 @@
+of products in this category are:
+
+* Apache Kafka
+* Apache Pulsar
+* Google Pubsub
+* AWS SNS
+* RabbitMQ
+
+Using an intermediate messaging system is in some cases preferred compared to
+a direct network connection between the telemetry producer and consumer.
+Reasons for using such an intermediate medium could be:
+
+* Protecting against backend failure
+* Security Policies
+* Network Policies
+* Buffering
+
+An extra motivation to have a consistent definition of the payload is that
+it would be easy to transfer the OTLP data from one messaging system to
+another just by reading the payload from one system and writing it to
+another without the need for transformations.
+
+## Explanation
+
+Because the OTLP payload that is sent over messaging systems is
+well-defined, it’s easy to implement new systems consistently.
+An implementation should at least support `otlp_proto`, meaning that the
+payload is Protobuf serialized from `ExportTraceServiceRequest` for traces,
+`ExportMetricsServiceRequest` for metrics, or `ExportLogsServiceRequest` for
+logs.
+
+Optionally an implementation could support `otlp_json` or other alternative
+encodings like jaeger or zipkin payloads. Alternative encodings should be configured
+by an `encoding` field in the configuration file.
+
+For the user it’s beneficial that both an exporter, and a receiver are
+implemented in the OpenTelemetry Collector, but SDK developers could
+also implement a language specific exporter.
+
+For log data, implementors of a receiver are encouraged to also support
+receiving raw data from a topic. This enables scenarios were
+non-OpenTelemetry log producers can produce a stream of logs, either
+structured or unstructured, on that topic and the receiver wraps the log
+data in a configurable resource.
+
+## Internal details
+
+The default implementation must implement `otlp_proto`, meaning that the
+payload is Protobuf serialized from `ExportTraceServiceRequest` for traces,
+`ExportMetricsServiceRequest` for metrics, or `ExportLogsServiceRequest` for
+logs. If an implementation support other encodings an `encoding` field should
+be added to the configuration to make it switchable.
+
+| value | trace | metric | log | description |
+|---|---|---|---|---|
+| `otlp_proto` (default) | X | X | X | protobuf serialized from the `Export(Trace/Metric/Log)ServiceRequest` |
+| `otlp_json` | X | X | X | proto3 json representation of a `Export(Trace/Metric/Log)ServiceRequest` |
+| `jaeger_proto` | X | - | - | the payload is deserialized to a single Jaeger proto `Span` |
+| `jaeger_json` | X | - | - | the payload is deserialized to a single Jaeger JSON Span using `jsonpb` |
+| `zipkin_proto` | X | - | - | the payload is deserialized into a list of Zipkin proto spans |
+| `zipkin_json` | X | - | - | the payload is deserialized into a list of Zipkin V2 JSON spans |
+| `zipkin_thrift` | X | - | - | the payload is deserialized into a list of Zipkin Thrift spans |
```

Comment:
I'm proposing changes to these in PR linked below. The types would be something like `zipkinv1/thrift`, `zipkinv2/protobuf`. Aside from the formatting it treats zipkinv1 and zipkinv2 as two separate protocols instead of putting them both behind `zipkin`.

Reply:
I think I'm going to remove the other encoding out of the spec as @tigrannajaryan  suggests